### PR TITLE
Implement a gameplay visual settings overlay

### PR DESCRIPTION
**Pull request by UselessToucan**
Creating PR since I need feedback.

I'm not sure about these things:
1. What to do with fail/pause overlays? It it ok to just add this VisualSettings object there too? I guess this is how it is implemented in osu! stable.
2. Is it acceptable to implement visual settings the same way as "Playback" thing which is used only for replays? For me it looks natural to implement it that way, but it is only my opinion.
3. What do you guys think about the logic behind auto hiding and time stopping?

If you are interested, here are a couple videos which shows appearance/behavior: [Gameplay](https://youtu.be/XQKnQHVtniw), [Replay](https://youtu.be/GcwjoMPyyeM)

### CODE REVIEW

**Review by smoogipoo — CHANGES_REQUESTED** [review]

File: osu.Game/Screens/Play/PlayerSettings/PlayerGroup.cs [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
 using osu.Game.Graphics.Sprites;
 using osu.Game.Graphics.UserInterface;
+using OpenTK;
+using OpenTK.Graphics;
 
-namespace osu.Game.Screens.Play.ReplaySettings
+namespace osu.Game.Screens.Play.PlayerSettings
 {
-    public abstract class ReplayGroup : Container
+    public abstract class PlayerGroup : Container
```
Comment (smoogipoo) [inline osu.Game/Screens/Play/PlayerSettings/PlayerGroup.cs]: Let's rename this to `PlayerSettingsGroup`.

**Comment by peppy** [comment]
Probably need @arflyte's input for this. Not sure if it should be separate from the global settings display, as opposed to just providing a default filter when in a gameplay screen.

**Comment by UselessToucan** [comment]
What's new:
Visual settings object which is created on the Player screen now behaves the same way as Playback settings. It is visible during replay and invisible during actual gameplay.
Another one visual settings was added to the PlayerLoader screen. It is available during gameplay. Please note that it prevents Player from loading while hovered.

[Gameplay video](https://www.youtube.com/watch?v=nq6pk16IK-4)
[Replay video](https://www.youtube.com/watch?v=RSiy5HthsxA)

**Comment by smoogipoo** [comment]
Dunno if you had a microphone spying on us while we talked about having it in PlayerLoader, but that was definitely a good move. A few comments:

1. It should be previewable during `PlayerLoader`.
2. It should be changeable from `PauseOverlay`, with the same sort of preview as osu-stable.

**Comment by peppy** [comment]
(i gave feedback in discord on that part)

**Comment by UselessToucan** [comment]
@smoogipoo What do you mean by "previewable"? Do you mean that expander should be collapsed when PlayerLoader screen is displayed for the first time?
About `PauseOverlay`: as far as I remember, @peppy wrote to keep it only on PlayerLoader screen for now.

**Comment by peppy** [comment]
By preview, he means that you should instantly be able to see the effect, possibly just while hovering.

**Comment by UselessToucan** [comment]
This [blur](https://github.com/ppy/osu/blob/master/osu.Game/Screens/Play/Player.cs#L385) and [dim](https://github.com/ppy/osu/blob/master/osu.Game/Screens/Play/Player.cs#L386) related magic is implemented inside of Player class. I planned to extract it in separate "code quality" PR later, but I can do it here.

**Comment by peppy** [comment]
I'd prefer a separate PR, personally. I'm fine with this being merged initially with no preview.

**Comment by UselessToucan** [comment]
@peppy @smoogipoo Could you please check this PR again and write what should be done in order to merge it?

**Review by smoogipoo — APPROVED** [review]
LGTM

**Pull request merged**